src/result_processor.c

```diff
@@ -1212,6 +1318,103 @@ static void addResultProcessor(AREQ *r, ResultProcessor *rp) {
   r->qiter.endProc = dummyHead.upstream;
 }
 
+// Insert the result processor before the first occurrence of a specific RP type in the upstream
+static bool addResultProcessorBeforeType(AREQ *r, ResultProcessor *rp, ResultProcessorType target_type) {
+  ResultProcessor *cur = r->qiter.endProc;
+  ResultProcessor dummyHead = { .upstream = cur };
+  ResultProcessor *downstream = &dummyHead;
+
+  bool found = false;
+
+  // Search for the target result processor type
+  while (cur) {
+    // Change downstream -> cur(type) -> cur->upstream
+    // To: downstream -> rp -> cur(type) -> cur->upstream
+    if (cur->type == target_type) {
+      rp->parent = &r->qiter;
```

Comment:
where are you using `&r->qiter`?
If needed, i suggest to move to the rp intiliztion as its not related to what this function does.

Reply:
I don't think we should "treat" this RP different in initialization from other RPs, even the debug ones (Timeout, Crash) don't initialize `rp->parent` and set the pointer at `addResultProcessor`.